### ISSUE TITLE
Adding Media to Notes

### ISSUE THREAD

**Issue by super-lurker** [maelchiotti/LocalMaterialNotes#104]
Most modern note-taking apps give you the ability to add audio recordings and images to your notes.
I hope to see this feature in the app someday. Otherwise, it's a solid note-taking app.
:)

**Comment by maelchiotti** [comment]
Hi @super-lurker, thank you, I'm glad you like the app!

The [editor library](https://github.com/fleather-editor/fleather) I use does support embedding images, so it shouldn't be hard to support this.

For audio recordings, I would need to develop the embeds myself, along with UI to record them, delete them etc., so I can't say if I'll ever give it a try.

**Comment by super-lurker** [comment]
I'll take what i can get, thanks :)

**Comment by maelchiotti** [comment]
Hi @super-lurker. I've looked into it and here are my troubles with adding images support:

- **Get the image:** an image can come from several sources (mainly clipboard, local file, camera and internet link). I might be hard to support all of them. I could support only some of them however (such as clipboard + local file).
- **Display the image:** the fleather library doesn't fully support images, it just allows you to create a custom embed widget in the editor. Thus, it would be up to me to figure out how to display images, regarding their size, their position... It could be hard to make it work for everyone and for every image.
- **Store the image:** in fleather, notes are stored as a JSON string. So I would either need to include the images as a string format (like base64) inside the notes (but I don't love this idea, adding lots of big images would make the JSON very big). I could also store the images somewhere, and reference the path to the image in the JSON.

To sum it up, it's not as straightforward as I initially thought. So I don't plan on adding this anytime soon, this will remain a text-based note taking app for now. I will however leave the issue opened in case I ever feel like tackling this issues!

**Comment by super-lurker** [comment]
local file is what most ppl will ever need, the rest is just bonus.
fleather sounds like a pain to work with :)

**Comment by maelchiotti** [comment]
Yes, I agree. If I work on it I will just begin with local files. I guess I could display all images at full width (with a max width for landscape or tablets). And if I manage to compress the images, maybe they won't be too heavy to store as base64. I'll take a second look soon if I find the time!

And fleather is actually really great! I tested like 3-4 editors before settling on this one. The maintainers are very nice as well. I just believe they don't have the time or resources to support more complex features out of the box. Maybe I'll switch to super_editor when they publish the promising new version of their editor if it makes this app more capable.
